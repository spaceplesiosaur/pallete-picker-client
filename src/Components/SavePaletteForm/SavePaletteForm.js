import React, { useState } from 'react';
import './SavePaletteForm.scss';
import { getData, postProject } from '../../apiCalls';
import { connect } from 'react-redux';
import { setAllProjects, setPalettes } from '../../actions';
import { bindActionCreators } from 'redux';

export const SavePaletteForm = ({ colorList, allSetProjects, setAllProjects, setPalettes }) => {
   const [paletteName, setPaletteName] = useState('');
   const [projectID, setProjectID] = useState('');

    const handleSubmit = async (e) => {
        e.preventDefault();
       postPalette();
       setPaletteName('');
    };

    const fetchProjectsAgain = async () => {
      const fetchedProjects = await getData('https://palette-picker-ac.herokuapp.com/api/v1/projects', 'projects');
      setAllProjects(fetchedProjects);
    };

    const fetchPalettesAgain = async () => {
      const fetchedPalettes = await getData('https://palette-picker-ac.herokuapp.com/api/v1/palettes', 'palettes');
      setPalettes(fetchedPalettes);
    };

    const postPalette = async () => {
        let completeColours = colorList.map((palette) => {
            return palette.color;
        });
        let palettes = {
            project_id: projectID,
            name: paletteName,
            color1: completeColours[0],
            color2: completeColours[1],
            color3: completeColours[2],
            color4: completeColours[3],
            color5: completeColours[4]
          };
      await postProject('https://palette-picker-ac.herokuapp.com/api/v1/palettes', palettes, 'palettes');
      await fetchPalettesAgain();
      await fetchProjectsAgain();
    };

    const displayProjects = allSetProjects.map(project => {
           return (
            <option key={project.id} value={project.id}>{project.name}</option>
           )
    });

    const chosenProject = (value) => {
      setProjectID(value)
    };
    
    const disableBtn = paletteName.length ? false : true;

    return (
       <form className='palette-form'>
           <select onChange={(e) => chosenProject(e.target.value)}>
               <option>Choose Project</option>
               {displayProjects}
           </select>
           <input type='text'
                  placeholder='Add Palette Name'
                  value={paletteName}
                  onChange={(e) => setPaletteName(e.target.value)}
            />
           <button disabled={disableBtn} onClick={handleSubmit}>Save Palette</button>
       </form>
    )
};
const mapStateToProps = ({allSetProjects}) => ({
    allSetProjects
})

const mapDispatchToProps = (dispatch) => (
  bindActionCreators({
    setAllProjects,
    setPalettes
  }, dispatch)
);
export default connect(mapStateToProps, mapDispatchToProps)(SavePaletteForm);
