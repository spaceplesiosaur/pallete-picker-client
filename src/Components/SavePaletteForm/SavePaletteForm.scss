 .palette-form {
    select{
        font-size: 20px;
        padding: 5px;
    }
    input {
        font-size: 20px;
        border: none;
        margin: 0 10px 0 10px;
        padding: 5px;
        border-radius: 10px;
    }
    button {
        font-size: 20px;
        padding: 5px;
        border-radius: 10px;
        background-color: #FCC606;
        cursor: pointer;
    }
 }