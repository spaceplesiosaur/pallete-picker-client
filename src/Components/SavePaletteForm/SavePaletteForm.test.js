import React from 'react';
import { shallow } from 'enzyme';
import SavePaletteForm from './SavePaletteForm'

describe('SavePaletteForm', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<SavePaletteForm />)
  })

  it('should render with the correct properites', () => {
    expect(wrapper).toMatchSnapshot();
  })
})
