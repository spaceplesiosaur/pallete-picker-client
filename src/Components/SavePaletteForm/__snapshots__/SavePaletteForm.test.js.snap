// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavePaletteForm should render with the correct properites 1`] = `
<form
  className="palette-form"
>
  <select
    onChange={[Function]}
  >
    <option>
      Choose Project
    </option>
    <option />
    <option />
  </select>
  <input
    onChange={[Function]}
    placeholder="Add Palette Name"
    type="text"
    value=""
  />
  <button
    disabled={true}
    onClick={[Function]}
  >
    Save Palette
  </button>
</form>
`;
